bp/src/BPConfiguration.m

```diff
@@ -140,12 +140,36 @@ typedef NS_OPTIONS(NSUInteger, BPOptionType) {
             "A .GlobalPreferences.plist simulator preferences file to be copied to any newly created simulators before booting"},
     {363, "script-file", BP_MASTER | BP_SLAVE, NO, NO, required_argument, NULL, BP_VALUE | BP_PATH, "scriptFilePath",
         "A script that will be called after the simulator is booted, but before tests are run. Can be used to do any setup (e.g. installing certs). The environment will contain $BP_DEVICE_ID with the ID of the simulator and $BP_DEVICE_PATH with its full path. Exit with zero for success and non-zero for failure."},
+    {364, "test-plan-path", BP_MASTER | BP_SLAVE, NO, NO, required_argument, NULL, BP_VALUE | BP_PATH, "testPlanPath", "The path of a json file which describes the test plan. It's a equivalent of .xctestrun file generated by an external build system, e.g. Bazel"},
 
     {0, 0, 0, 0, 0, 0, 0}
 };
 
 static NSUUID *sessionID;
 
+@implementation BPTestPlan
+
+- (BOOL)isValid {
```

Comment:
Related to my comment below of giving a better error, we could pass an `NSError` here and say _why_ the plan is invalid so that when we fail we tell the user what to fix.